src/include/daos/task.h

```diff
@@ -145,8 +145,8 @@ dc_obj_fetch_shard_task_create(daos_handle_t oh, daos_handle_t th,
 
 int
 dc_obj_fetch_task_create(daos_handle_t oh, daos_handle_t th, uint64_t flags,
-			 daos_key_t *dkey, unsigned int nr,
-			 daos_iod_t *iods, d_sg_list_t *sgls,
+			 daos_key_t *dkey, unsigned int nr, daos_iod_t *iods,
+			 d_sg_list_t *sgls, void *extra_args,
```

Comment:
extra args is to pass in the fail_info for the fetch task for data recovery, it need to know the previous failure info to decide fetch from which set of targets.
Do you have better idea to pass that info to the recovery fetch task?

Reply:
I thought you can use push/pop to attach the variable to the task? 

Btw: ec_degrade_fetch probably should be done explicitly in a separate function, which will
1. Analyse the failure information got from obj_comp_cb(), and decide the failure shard to be rebuilt.
2. Issue simple fetch task as other shard fetch.
3. reconstruct the data inside this function.

Instead of burying these logics inside the object I/O stack, not easy to follow.

Reply:
push/pop need to be called in pair (one push and one pop), and it is not very convenient to use when create/schedule a new recovery task.

I considered to use another separate func as you suggested (ec_degrade_fetch things) to do the recovery task, but later found that what the recovery task need to do basically is same as normal fetch task, just with some additional ISAL recovery work. So I reused the existing fetch task and reuse most of the code there.